### PR TITLE
Add a BBE for WebSocket service error handling

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
$subject.

### CODE REVIEW

File: examples/websocket-service-error-handling/websocket_service_error_handling.bal [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        return new ChatService();
+    }
+}
+
+service class ChatService {
+    *websocket:Service;
+
+    remote function onMessage(websocket:Caller caller, string chatMessage) returns error? {
+        io:println(chatMessage);
+        // If an error occurs, error is returned from the `onMessage`. It is printed on the
```
Comment (praneesha) [inline examples/websocket-service-error-handling/websocket_service_error_handling.bal:22]:
```suggestion
        // If an error occurs, the error is returned from the `onMessage`. It is printed on the
```

File: examples/websocket-service-error-handling/websocket_service_error_handling.bal [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        // server terminal along with the stack trace
```
Comment (praneesha) [inline examples/websocket-service-error-handling/websocket_service_error_handling.bal:23]:
```suggestion
        // server terminal along with the stack trace.
```

File: examples/websocket-service-error-handling/websocket_service_error_handling.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# WebSocket service - Error handling
+
+The Ballerina `websocket` module allows returning `error`s from the `get` resource and `remote` methods. In the initial WebSocket upgrade service, you can do validations like origin checking and cancel the WebSocket handshake by returning an error from the `get` resource if the validation fails. Once the connection is upgraded to a WebSocket connection if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace. Use `do on fail` to catch and handle the error.
```
Comment (praneesha) [inline examples/websocket-service-error-handling/websocket_service_error_handling.md:3]:
```suggestion
The Ballerina `websocket` module allows returning errors from the `get` resource and `remote` methods. In the initial WebSocket upgrade service, you can do validations like origin checking and cancel the WebSocket handshake by returning an error from the `get` resource if the validation fails. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace. Use `do on fail` to catch and handle the error.
```

File: examples/websocket-service-error-handling/websocket_service_error_handling.metatags [hunk 4]
```diff
@@ -0,0 +1,2 @@
+description: BBE on error handling in WebSocket service.
```
Comment (praneesha) [inline examples/websocket-service-error-handling/websocket_service_error_handling.metatags:1]:
```suggestion
description: This BBE demonstrates error handling in WebSocket services.
```

File: examples/websocket-service-error-handling/websocket_service_error_handling.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# WebSocket service - Error handling
+
+The Ballerina `websocket` module allows returning errors from the `get` resource and `remote` methods. In the initial WebSocket upgrade service, you can do validations like origin checking and cancel the WebSocket handshake by returning an error from the `get` resource if the validation fails. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace. Use `do on fail` to catch and handle the error.
```
Comment (shafreenAnfar) [inline examples/websocket-service-error-handling/websocket_service_error_handling.md:3]: The Ballerina `websocket` module allows returning errors from the `get` resource and `remote` methods. Initial WebSocket upgrade service can do various validations. For instance, origin checking and canceling the WebSocket handshake by returning an error from the `get` resource if the validation fails. In this case it results in `xxx xxxxxxxxxxxxx` HTTP response. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace and it continues to work. Use `do/fail` to catch and handle the error when needed.

File: examples/websocket-service-error-handling/websocket_service_error_handling.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# WebSocket service - Error handling
+
+The Ballerina `websocket` module allows returning errors from the `get` resource and remote methods. The initial WebSocket upgrade service can do various validations. For instance, origin checking and canceling the WebSocket handshake by returning an `error` from the get resource if the validation fails. In this case, it results in a `400 Bad Request` HTTP response in generic cases. If the returned error is a `websocket:AuthzError`, or a `websocket:AuthnError`, `403 Forbidden` and `401 Unauthorized` HTTP responses will be returned respectively. Once the connection is upgraded to a WebSocket connection if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack trace and it continues to work. Use do/fail to catch and handle the error when needed.
```
Comment (shafreenAnfar) [inline examples/websocket-service-error-handling/websocket_service_error_handling.md:3]: The Ballerina `websocket` module allows returning errors from the `get` resource and remote methods. The initial WebSocket upgrade service can do various validations. For instance, origin checking and canceling the WebSocket handshake by returning an `error` from the get resource if the validation fails. In this case, it results in a `400 Bad Request` HTTP response, which is the default response. If the returned error is a `websocket:AuthzError`, or a `websocket:AuthnError`, `403 Forbidden` and `401 Unauthorized` HTTP responses are returned respectively. Once the connection is upgraded to a WebSocket connection, if an error is returned from a remote method in the `websocket:Service`, the error is printed on the server terminal along with the stack-trace and it continues to receive messages. Use `do/fail` block to catch and handle the error when needed.

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**